ledokku-bootstrap.sh

```diff
@@ -1,52 +1,187 @@
+      echo "You chose to update your system"
+      # Update and skip to next step
+      echo "${YELLOW}Updating System${END}"
+      sudo dpkg --configure -a
+      sudo apt -y --purge autoremove
+      sudo apt install -f
+      sudo apt -y update
+      wait
+      echo "${YELLOW}Upgrading System${END}"
+      sudo apt -y upgrade
+      sudo apt -y autoclean
+      sudo apt -y --purge autoremove &
+      process_id=$!
+      wait $process_id
+      echo "Exit status: $?"
+  else
+      echo "${YELLOW}You chose to skip.${END}"
+      # Should skip to next step
+  fi
+}
+
+function ensure-dokku() {
+  
+  # Confirming the existance of Dokku
+  ## If exists, promt for update
+  ## If not exits Let him download.
+  if which dokku >/dev/null; then
+      echo "${GREEN}Dokku Exists${END}"
+      # Checking Dokku vesrion and comparing it with the latest Version
+      # In case of version changes in dokku, we need to change this varibale: LATEST_DOKKU_VERSION.
+      # We can also rename the variable => LATEST_DOKKU_VERSION to PREFFERED_DOKKU_VERSION
+      
+      EXISTING_DOKKU_VERSION="$(dokku version)"
+      LATEST_DOKKU_VERSION="dokku version 0.24.10"
+      
+      if [[ "$EXISTING_DOKKU_VERSION" == "$LATEST_DOKKU_VERSION" ]]; then
+        echo  "${YELLOW}Dokku is upto date${END}"
+      else
+        whiptail --title "Warning !!" --msgbox "Read carefully before proceeding:\n
+        You are currently using "$EXISTING_DOKKU_VERSION"
+        but the latest was "$LATEST_DOKKU_VERSION" \n\nIn the next dialog box, you can update your dokku or skip \n\nFor more info check the dokku CHANGELOG before doing the upgrade: https://github.com/dokku/dokku/releases" 20 60
+        # Promt for update
+        if (whiptail --title "Updating Dokku" --yes-button "Update" --no-button "Skip"  --yesno "Would you like to update your Dokku?" 10 60) then
+            echo "${YELLOW}You chose Update.${END}"
+            # Update Dokku
+            echo "${YELLOW}Upgrading Dokku${END}"
+            sudo apt-get -y update -qq
+            wait
+            sudo apt-get -qq -y --no-install-recommends install dokku herokuish sshcommand plugn gliderlabs-sigil dokku-update dokku-event-listener
+            wait
+            sudo apt -y upgrade &
+            process_id=$!
+            wait $process_id
+            echo "Exit status: $?"
+            echo "${YELLOW}Updated to ${GREEN} $LATEST_DOKKU_VERSION ${END}"
+            # Dokku Updated
+        else
+            echo "${YELLOW}You chose to skip dokku updates.${END}"
+            # Dokku Update skipped
+        fi
+      fi
+  else
+      echo "${RED}Dokku does not exist${END}"
+      # Show messagebox and make it mandatory to download and install dokku
+      whiptail --title "Unable to Detect Dokku" --msgbox "If you want to insatll your app using ledokku, it is madatory to install Dokku. So, I would like to install Dokku on behalf of you." 10 60
```

Comment:
@akhil-naidu typo in the first `install`

Reply:
Edit 01: I will take care of that typo.

let me also know about other issues.